lib/web_ui/lib/ui_web/src/ui_web/flutter_views_proxy.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:js_interop';
+import 'package:ui/src/engine.dart';
+
+/// Exposes web-only functionality for this app's `FlutterView`s objects.
+final FlutterViewManagerProxy views = FlutterViewManagerProxy(
+  viewManager: EnginePlatformDispatcher.instance.viewManager,
+);
+
+/// This class exposes web-only attributes for the registered views in a multi-view app.
+class FlutterViewManagerProxy {
+  FlutterViewManagerProxy({
+      required FlutterViewManager viewManager,
+    }) : _viewManager = viewManager;
+  // The proxied viewManager instance.
+  final FlutterViewManager _viewManager;
+  /// Returns the `initialData` configuration value passed from JS when `viewId` was added.
+  ///
+  /// Developers can access the initial data from Dart in two ways:
+  ///  * Defining their own `staticInterop` class that describes what you're
+  ///    passing to your views, to retain type safety on Dart (preferred).
+  ///  * Calling [NullableUndefineableJSAnyExtension.dartify] and accessing the
+  ///    returned object as if it were a [Map] (not recommended).
+  Object? getInitialData(int viewId) {
+    return _viewManager.getOptions(viewId)?.initialData;
```

Comment:
I guess the backing in the viewManager needs to be `Map<int, Object>` so we can store a JSAny (or any other Dart object)